pkg/security/security.go

```diff
@@ -235,3 +244,14 @@ type SecretItem struct {
 
 	ExpireTime time.Time
 }
+
+type CredFetcher interface {
```

Comment:
Still not clear why we need this interface - and not use the gRPC interface for getting tokens ( and associated implementations )

Reply:
Discussed offline. gRPC interface may not be used directly here. Post 1.7, I will evaluate the possibility of wrapping the gRPC library implementation in this API.